### PR TITLE
fixes to the asset CRUD and refactor

### PR DESCRIPTION
**Pull request by FBRTMaka**
@oceanzus This will start using the redis / celery services

@DanielJMaher we'll be able to sort the asset management page based on deployment type (PAST or PRESENT) which seems like a easy and BIG win.

@asascience-open/team-ooi-ui , make sure you get your redis / celery worked out!!

### CODE REVIEW

**Comment by oceanzus** [comment]
@FBRTMaka What happens if redis and/or celery services are not running?

**Comment by FBRTMaka** [comment]
@oceanzus   If celery is not running then there's no problem.  If redis isn't running then there won't be a cache service -- basically the services will fail on loading the cache.

File: ooiservices/app/config.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ COMMON: &common
     HOST: localhost
     PORT: 4000
     # UFrame config
-    UFRAME_URL:             'http://localhost:12575'
+    UFRAME_URL:             'http://localhost:12576'
```
Comment (birdage) [inline ooiservices/app/config.yml]: :+1:

File: ooiservices/app/uframe/assetController.py [hunk 2]
```diff
@@ -6,24 +6,98 @@
...
+                for events in row['events']:
+                    if events['locationLonLat'] is not None and\
+                            lat == 0.0 and lon == 0.0:
+                        lat = events['locationLonLat'][1]
+                        lon = events['locationLonLat'][0]
+                    if events['class'] == '.DeploymentEvent':
+                        has_deployment_event = True
+                row['hasDeploymentEvent'] = has_deployment_event
+                row['coordinates'] = convert_lat_lon(lat, lon)
+                lat = 0.0
```
Comment (birdage) [inline ooiservices/app/uframe/assetController.py]: still resetting lat/lon to 0.0?

File: ooiservices/app/uframe/assetController.py [hunk 3]
```diff
@@ -6,24 +6,98 @@
...
+                if len(row['ref_des']) == 27:
+                    row['asset_class'] = '.InstrumentAssetRecord'
+                if len(row['ref_des']) < 27:
+                    row['asset_class'] = '.AssetRecord'
+
+                if deployment_number is not None:
+                    row['deployment_number'] = deployment_number
+                for events in row['events']:
+                    if events['locationLonLat'] is not None and\
+                            lat == 0.0 and lon == 0.0:
```
Comment (birdage) [inline ooiservices/app/uframe/assetController.py]: not sure i understand the 0.0 here.

File: ooiservices/app/uframe/assets.py [hunk 4]
```diff
@@ -332,56 +213,61 @@ def create_asset():
...
+            data['@class'] = data.pop('asset_class')
+
+        print data
+        print uframe_assets_url
+        print _uframe_headers()
+        response = requests.post(uframe_assets_url,
+                                 data=json.dumps(data),
+                                 headers=_uframe_headers())
+
+        print response.text
```
Comment (birdage) [inline ooiservices/app/uframe/assets.py]: do we want to remove the prints? here and above?

File: ooiservices/app/__init__.py [hunk 5]
```diff
@@ -21,7 +21,7 @@
 login_manager = LoginManager()
 login_manager.session_protection = 'strong'
 
-cache = Cache(config={'CACHE_TYPE': 'simple'})
+cache = Cache(config={'CACHE_TYPE': 'redis'})
```
Comment (birdage) [inline ooiservices/app/__init__.py]: should this config option be in the config file?

File: ooiservices/app/uframe/assets.py [hunk 6]
```diff
@@ -438,7 +325,8 @@ def get_asset_classes_list():
         if row['class'] is not None:
             data.append(row['class'])
     data = _remove_duplicates(data)
-    return jsonify({ 'class_types' : data })
+    return jsonify({'class_types': data})
+
 
 @cache.memoize(timeout=3600)
```
Comment (birdage) [inline ooiservices/app/uframe/assets.py]: this uses a different cache timeout?

**Pull request merged**